src/main/java/wooteco/subway/line/dao/LineDao.java

```diff
@@ -0,0 +1,112 @@
+package wooteco.subway.line.dao;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Map;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import wooteco.subway.exception.DuplicationException;
+import wooteco.subway.exception.NotFoundException;
+import wooteco.subway.line.model.Line;
+
+@Repository
+public class LineDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public LineDao(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public Line save(Line line) {
+        validateDuplicateNameAndColor(line);
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        String sql = "INSERT INTO line (`name`, color) VALUES (?, ?)";
+        jdbcTemplate.update(connection -> {
+            PreparedStatement preparedStatement = connection
+                .prepareStatement(sql, new String[]{"id", "name", "color"});
+            preparedStatement.setString(1, line.getName());
+            preparedStatement.setString(2, line.getColor());
+            return preparedStatement;
+        }, keyHolder);
+
+        Map<String, Object> keys = keyHolder.getKeys();
+        return new Line((Long) keys.get("id"), (String) keys.get("name"),
```

Comment:
```suggestion
       return findLineById(keyHoder.getKey());
```
는 어떻게 생각하나요 키 홀더에서 키 하나만 응답하는 메소드는 확실치 않네요 😭

Reply:
확실히 말씀하신 대로 수정하니, 형변환을 해줄 필요도 없고 코드도 깔끔해져 보기가 좋네요 👍

그런데, 제 생각엔 이 방법을 사용하게 되면 기존의 방법과 달리 Line을 find 하게 되는 과정에서 Connection이 한번 더 일어나날 것 같습니다. 만약 그렇다면 (미약한 수준일지라도) 성능 이슈가 생길 것 같은데, 이 정도는 감수하고 갈만 한걸까요?